kubectl-plugin/pkg/cmd/get/get_workergroup.go

```diff
@@ -0,0 +1,231 @@
+
+	resTable := &v1.Table{
+		ColumnDefinitions: []v1.TableColumnDefinition{
+			{Name: "Name", Type: "string"},
+			{Name: "Replicas", Type: "string"},
+			{Name: "CPUs", Type: "string"},
+			{Name: "GPUs", Type: "string"},
+			{Name: "TPUs", Type: "string"},
+			{Name: "Memory", Type: "string"},
+		},
+	}
+
+	for _, workerGroup := range workerGroupSpecs {
+		podList, err := k8sClient.KubernetesClient().CoreV1().Pods(namespace).List(ctx, v1.ListOptions{
+			LabelSelector: fmt.Sprintf("ray.io/cluster=%s", workerGroup.GroupName),
+		})
+		if err != nil {
+			return err
+		}
+
+		readyWorkerReplicas := calculateReadyReplicas(*podList)
+
+		workerGroupResources := calculateDesiredResourcesForWorkerGroup(workerGroup)
+
+		gpus := workerGroupResources[corev1.ResourceName(util.ResourceNvidiaGPU)]
+		tpus := workerGroupResources[corev1.ResourceName(util.ResourceGoogleTPU)]
+
+		resTable.Rows = append(resTable.Rows, v1.TableRow{
+			Cells: []interface{}{
+				workerGroup.GroupName,
+				fmt.Sprintf("%d/%d", readyWorkerReplicas, *workerGroup.Replicas),
+				workerGroupResources.Cpu().String(),
+				gpus.String(),
+				tpus.String(),
+				workerGroupResources.Memory().String(),
+			},
+		})
+	}
+
+	return resultTablePrinter.PrintObj(resTable, output)
+}
+
+func calculateDesiredResourcesForWorkerGroup(workerGroupSpec rayv1.WorkerGroupSpec) corev1.ResourceList {
+	if workerGroupSpec.Suspend != nil && *workerGroupSpec.Suspend {
+		return corev1.ResourceList{}
+	}
+
+	podResource := calculatePodResource(workerGroupSpec.Template.Spec)
+	totalResource := corev1.ResourceList{}
+
+	for range *workerGroupSpec.Replicas {
+		for name, quantity := range podResource {
+			totalResource[name] = quantity.DeepCopy()
+			var quantity resource.Quantity = totalResource[name]
+			(&quantity).Mul(int64(*workerGroupSpec.Replicas))
+		}
+	}
+
+	return totalResource
+}
+
+// calculatePodResource returns the total resources of a pod.
+// Request values take precedence over limit values.
+func calculatePodResource(podSpec corev1.PodSpec) corev1.ResourceList {
+	podResource := corev1.ResourceList{}
+	for _, container := range podSpec.Containers {
+		containerResource := container.Resources.Requests
+		if containerResource == nil {
+			containerResource = corev1.ResourceList{}
+		}
+		for name, quantity := range container.Resources.Limits {
+			if _, ok := containerResource[name]; !ok {
+				containerResource[name] = quantity
+			}
+		}
+		for name, quantity := range containerResource {
+			if totalQuantity, ok := podResource[name]; ok {
+				totalQuantity.Add(quantity)
+				podResource[name] = totalQuantity
+			} else {
+				podResource[name] = quantity
+			}
+		}
+	}
+	return podResource
+}
+
+// calculateReadyReplicas calculates ready worker replicas at the cluster level
+// A worker is ready if its Pod has a PodCondition with type == Ready and status == True
+func calculateReadyReplicas(pods corev1.PodList) int32 {
```

Comment:
copied from https://github.com/ray-project/kuberay/blob/bd53766155407fad258dbcb5dd5375281047699f/ray-operator/controllers/ray/utils/util.go#L387 to avoid some weird errors if we try to import and use that func directly.

Reply:
Is there any way to be DRY and reuse this function without the errors?